### ISSUE TITLE
progressive reveal for select tags

### ISSUE THREAD

**Issue by danielepolencic** [UKHomeOffice/passports-frontend-toolkit#50]
This refers to https://github.com/UKHomeOffice/passports-frontend-toolkit/pull/47.

**Comment by danielepolencic** [comment]
**TL;DR: We decided to not merge the PR since it doesn't promote good practises.**

Rather than using showing/hiding an element based on a drop down selection, we decided to reorganise the step and break it down into several steps.

Consider the following scenario.

![image](https://cloud.githubusercontent.com/assets/1130499/17663217/d5bc7818-62e2-11e6-9703-8d58eb7e5e2e.png)

1. a user land on a page and is asked if they have a valid visa or residence permit.
2. the user has a valid permit.
3. but it's none of the options in the first dropdown (selecting any of these options fast-forwards the application to the next step).
4. a new drop down appears, with a similar (same) question but a different list of options
5. the user doesn't have any of the permits listed (selecting any of these options makes the user ineligible).
6. a free text field appears. Any input is valid and takes the user to the next step.

This page has few issues:

- when Javascript is disabled, the user can see all dropdowns and input fields, even if they have a valid permit from the first dropdown. Indeed the conditional reveal works only when JS is enabled.
- there're two drop downs that are similar (same questions), but different answers.
- The flow is very confusing: a user can successfully pass this step by choosing any options but *"I don't have any of these"* from the first dropdown OR by getting to the free text input field at the end. Users are not eligible for an RT when they pick any option but *"I don't have any of these"* from the second dropdown.

How one should rework this page:

![image](https://cloud.githubusercontent.com/assets/1130499/17669091/260a0576-6304-11e6-8767-d972b06af9c2.png)

The single page is split in two steps:

1. If you have a permit, you're immediately asked what kind of permit you have.
    - Any option from the list takes the user to the next step in the process
    - When users select *"None of the above"* they are taken to an intermediate page.
2. The intermediate page asks a similar (same) question, but with a different set of answers:
   - Any option from the list takes the user to the ineligible page.
   - When users select *"None of the above"* a free text field appears. Submitting the form takes the user to the next step in the process.

**Comment by sulthan-ahmed** [comment]
Fantastic suggestion.

Small suggestion no need to do but it would have been nicer if the screenshots of the suggested improvement had the actual radio options (e.g. Tier1, Tier 5 etc) rather than option 1.  I was confused until I read your conclusion at the end

**Comment by danielepolencic** [comment]
@sulthan-ahmed updated 👍

**Comment by sulthan-ahmed** [comment]
👍

**Comment by gavboulton** [comment]
Can this be closed now?

**Comment by danielepolencic** [comment]
I think so.
